controllers/admin_controller.go

```diff
@@ -0,0 +1,187 @@
+package controllers
+
+import (
+	"bcc_university/models"
+	"bcc_university/utils"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/kamva/mgm/v3"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func AddUserToClass(c *gin.Context) {
+	var payload = c.Param("classId")
+	class, err := GetClass(payload)
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": err.Error()})
+	}
+
+	var userPayload = c.Param("userId")
+	user := GetUser(userPayload)
+
+	if utils.Contains(user.ClassesEnrolled, class.ID) {
+		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": "user already participated in this class"})
+		return
+	}
+
+	if err = utils.ArrayMethod(&models.Class{}, class.ID, "$push", "participants", user.ID); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": err.Error()})
+		return
+	}
+
+	if err = utils.ArrayMethod(&models.User{}, user.ID, "$push", "classes_enrolled", class.ID); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": err.Error()})
+		return
+	}
+
+	user.RemSks -= class.Sks
+	if user.RemSks < 0 {
+		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": "cannot add more class"})
+		return
+	}
+
+	err2 := mgm.Coll(&models.User{}).Update(user)
+	err3 := mgm.Coll(&models.Class{}).Update(class)
+	if err2 != nil || err3 != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": err2.Error()})
```

Comment:
this shorthanded error handling could lead to a panic if err3 is not nil and the err2 is nil
```suggestion
	err2 := mgm.Coll(&models.User{}).Update(user)
	if err2 != nil {
		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": err2.Error()})
	err3 := mgm.Coll(&models.Class{}).Update(class)
	if err3 != nil {
		c.JSON(http.StatusBadRequest, gin.H{"status": "fail", "message": err3.Error()})
```